src/imageupload/imageuploadcommand.js

```diff
@@ -18,6 +18,23 @@ import Command from '@ckeditor/ckeditor5-core/src/command';
  * @extends module:core/command~Command
  */
 export default class ImageUploadCommand extends Command {
+	/**
+	 * @inheritDoc
+	 */
+	refresh() {
+		const model = this.editor.model;
+		const selection = model.document.selection;
+		const schema = model.schema;
+		const position = selection.getFirstPosition();
+		let parent = position.parent;
+
+		if ( parent != parent.root ) {
```

Comment:
I'm missing a comment why this check and parent swapping.

Reply:
ps.: Also this doesn't look consistent as it might swap to:

- blockQuote  (from p, h1, l1, etc inside blockquote)
- tableRow (from tableCell) - I'm not sure what should be there (table/tableCell - oncoming block content in table cell)
- $root (from p, h1, l1, etc)

Maybe it should lookup for allowed parent or something other in the schema?

Reply:
ps.: A comment would be sufficient also :)

Reply:
TBH, I copy&pasted it from the [`InsertTableCommand`](https://github.com/ckeditor/ckeditor5-table/blob/f03a755eb564d0e15285002664eb8b70af426829/src/commands/inserttablecommand.js#L73-L78) implementation without any deeper analysis. The upload command never splits the block, it inserts the new image next to the one the selection is anchored to. So I guess the `if` is correct here because it comes down to 

> get the first ancestor of the current block, unless it's the root

Reply:
BTW, have you seen https://github.com/ckeditor/ckeditor5-image/issues/225#issuecomment-407021363? Do you have any idea how to include a fix for that without breaking the multi-upload?

Reply:
I'll check that and let you know :)

Reply:
@oleq Sorry for longer reply. So I've manage to make #227 work by adding multi-file upload support to the `FileUploadCommand` itself:
```js
// Extend current behavior of uploading one file
const filesToUpload = Array.isArray( options ) ? options : [ options ];

for ( const { file, insertAt } of filesToUpload ) {
	uploadImage( editor, fileRepository, file, insertAt, doc );
}
```

Where `uploadImage()` is the method that previously uploaded a file.

This way the command can be disabled when an image is selected but it will allow UI to pass multiple files to upload at once - not as it is currently where each file is processed individually by a command. The problem is that after `execute()` the first image is selected and the command itself was disabled so next files were discarded as the command goes disabled then.

I have no other idea for solving this as probably the expected behavior is to select newly inserted image.

Reply:
ps.: the `ImageUploadUI` was modified to:
```js
view.on( 'done', ( evt, files ) => {
	const toUpload = [];

	for ( const file of Array.from( files ) ) {
		const insertAt = findOptimalInsertionPosition( editor.model.document.selection );

		if ( isImageType( file ) ) {
			toUpload.push( { file, insertAt } );
		}
	}

	editor.execute( 'imageUpload', toUpload );
} );
```
but those two snippets must be checked if there are OK as the files are inserted in revers order (AFAICS).